File: Taxi.java
public class Taxi extends Vehicle {

    private String driverName;

    public Taxi(int vehicleId, String vehicleNumber, int capacity, String driverName) {

        super(vehicleId, vehicleNumber, capacity);
        this.driverName = driverName;
    }

    @Override
    public void displayVehicleDetails() {

        System.out.println("\n----------- Taxi Details -----------");
        System.out.println("Vehicle ID : " + vehicleId);
        System.out.println("Vehicle Number : " + vehicleNumber);
        System.out.println("Capacity : " + capacity);
        System.out.println("Driver Name : " + driverName);
    }
}

File: c#-20260820T054914Z-1-001/c#/Taxi.cs
// Inheritance
// Taxi class inherits properties like vehicleId, vehicleNumber,
// and capacity from the Vehicle class.

public class Taxi : Vehicle
{
    // Taxi class has its own property as driverName
    private string driverName;

    public Taxi(int vehicleId, string vehicleNumber, int capacity, string driverName)
        : base(vehicleId, vehicleNumber, capacity)
    {
        this.driverName = driverName;
    }

    // Method overriding
    public override void displayVehicleDetails()
    {
        Console.WriteLine("\n----------- Taxi Details -----------");
        Console.WriteLine("Vehicle ID : " + vehicleId);
        Console.WriteLine("Vehicle Number : " + vehicleNumber);
        Console.WriteLine("Capacity : " + capacity);
        Console.WriteLine("Driver Name : " + driverName);
    }
}